src/quick-lint-js/parse.h

```diff
@@ -1441,12 +1441,41 @@ class parser {
     QLJS_WARNING_IGNORE_GCC("-Wshadow-local")
 
     std::optional<identifier> last_ident;
+    const char8 *async_token_begin = nullptr;
     function_attributes method_attributes = function_attributes::normal;
+    bool async_static = false;
 
     auto parse_and_visit_field_or_method_impl =
-        [this, &v](std::optional<identifier> property_name,
-                   source_code_span property_name_span,
-                   function_attributes method_attributes) -> void {
+        [this, &v, &async_static, &async_token_begin](
+            std::optional<identifier> property_name,
+            source_code_span property_name_span,
+            function_attributes method_attributes) -> void {
+      if (async_static && !this->peek().has_leading_newline) {
```

Comment:
What is the purpose of `!this->peek().has_leading_newline`? It looks like `!this->peek().has_leading_newline` would cause the following examples to parse differently. Am I correct?
```js
class C {
  async static m() {}
}
```
```js
class C {
  async static m
    () {}
}
```

Reply:
I'm not sure what the purpose of this new line check is anymore.